pkg/proxy/userspace/proxier.go

```diff
@@ -136,40 +139,50 @@ func IsProxyLocked(err error) bool {
 // if iptables fails to update or acquire the initial lock. Once a proxier is
 // created, it will keep iptables up to date in the background and will not
 // terminate if a particular iptables call fails.
-func NewProxier(loadBalancer LoadBalancer, listenIP net.IP, iptables iptables.Interface, pr util.PortRange, syncPeriod time.Duration) (*Proxier, error) {
+func NewProxier(loadBalancer LoadBalancer, listenIP net.IP, iptables iptables.Interface, pr util.PortRange, recorder record.EventRecorder, syncPeriod time.Duration, nodeRef *api.ObjectReference) (*Proxier, error) {
 	if listenIP.Equal(localhostIPv4) || listenIP.Equal(localhostIPv6) {
+		recorder.Eventf(nodeRef, "ErrProxyOnLocalhost", "Cannot proxy on localhost")
```

Comment:
Do we need an event here - the caller is going to fail.  We don't, in general, log app errors as events.  We might want to but I think that deserves its own design doc :)